src/dhcpmon/src/dhcp_device.h

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file dhcp_device.h
+ *
+ *  device (interface) module
+ */
+
+#ifndef DHCP_DEVICE_H_
+#define DHCP_DEVICE_H_
+
+#include <pcap.h>
+#include <stdint.h>
+#include <pthread.h>
+#include <net/if.h>
+#include <netinet/in.h>
+#include <net/ethernet.h>
+
+/** packet direction */
+typedef enum
+{
+    DHCP_RX,    /** RX DHCP packet */
+    DHCP_TX,    /** TX DHCP packet */
+
+    DHCP_DIR_COUNT
+} dhcp_packet_direction_t;
+
+/** dhcp health status */
+typedef enum
+{
+    DHCP_MON_STATUS_HEALTHY,        /** DHCP relay is healthy */
+    DHCP_MON_STATUS_UNHEALTHY,      /** DHCP relay is unhealthy and is missing out on some packets */
+    DHCP_MON_STATUS_INDETERMINATE,  /** DHCP relay health could not be determined */
+} dhcp_mon_status_t;
+
+/** DHCP device (interface) health counters */
+typedef struct
+{
+    uint64_t discovery;     /** DHCP discovery packets */
```

Comment:
s/discovery/discover/

The first packet of the handshake is technically called a "Discover" packet (no "y").

Reply:
done